pkg/operation/botanist/vpnshoot.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/charts"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/kubeapiserver"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/vpnseedserver"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/vpnshoot"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+)
+
+// DefaultVPNShoot returns a deployer for the VPNShoot
+func (b *Botanist) DefaultVPNShoot() (vpnshoot.Interface, error) {
+	var (
+		imageName         = charts.ImageNameVpnShoot
+		nodeNetwork       = b.Shoot.GetInfo().Spec.Networking.Nodes
+		nodeNetworkCIDR   string
+		reversedVPNValues *vpnshoot.ReversedVPNValues = &vpnshoot.ReversedVPNValues{
+			Enabled: false,
+		}
+	)
+
+	if nodeNetwork != nil {
+		nodeNetworkCIDR = *nodeNetwork
+	}
+
+	if b.Shoot.ReversedVPNEnabled {
+		imageName = charts.ImageNameVpnShootClient
+
+		reversedVPNValues = &vpnshoot.ReversedVPNValues{
+			Enabled:     true,
+			Header:      "outbound|1194||" + vpnseedserver.ServiceName + "." + b.Shoot.SeedNamespace + ".svc.cluster.local",
+			Endpoint:    b.outOfClusterAPIServerFQDN(),
+			OpenVPNPort: 8132,
+		}
+	}
+
+	image, err := b.ImageVector.FindImage(imageName, imagevector.RuntimeVersion(b.ShootVersion()), imagevector.TargetVersion(b.ShootVersion()))
+	if err != nil {
+		return nil, err
+	}
+
+	values := vpnshoot.Values{
+		Image:             image.String(),
+		VPAEnabled:        b.Shoot.WantsVerticalPodAutoscaler,
+		ReversedVPNValues: *reversedVPNValues,
```

Comment:
```suggestion
		ReversedVPNValues: reversedVPNValues,
```

Why did you make `reversedVPNValues` a pointer when you dereference it here again? 😕